packages/nx/src/hasher/git-hasher.ts

```diff
@@ -105,7 +105,7 @@ async function spawnProcess(
 async function getStagedFiles(path: string) {
   const { stdout: staged } = await spawnProcess(
     'git',
-    ['ls-files', '-s', '-z', '--exclude-standard', '.'],
+    ['ls-files', '--recurse-submodules', '-s', '-z', '--exclude-standard', '.'],
```

Comment:
@wickedest for the other 2 methods `getUnstagedFiles` and `getUntrackedFiles` I had the same `git` error at the begging when I just used the `--recurse-submodules` flag and therefore I've used the older command style which is the `git submodule foreach` and that worked fine. So if the same fix was applied to `getStagedFiles` it will work for older `git` versions as well.